### PR TITLE
template and schema for deploying vpn with auto subnet network

### PR DESCRIPTION
**Pull request by crcsmnky**
Googler (parikhs@) submitting a DM example template/schema for automating Cloud VPN deployment with auto subnet networks.

### CODE REVIEW

**Comment by crcsmnky** [comment]
No need for CLA, I'm a Googler (parikhs@).

File: examples/v2/vpn_auto_subnet/vpn-auto-subnet.jinja.schema [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    description: Shared secret for the peer VPN gateway
+
+  sourceRanges:
+    type: string
+    description: IPv4 range of allowed traffic sources from peer network
+
+  routeTag:
+    type: string
+    default: default-route
+    description: Tag name for route for outgoing traffic, use with instance tags
```
Comment (aljim) [inline examples/v2/vpn_auto_subnet/vpn-auto-subnet.jinja.schema]: Add new line end of file

File: examples/v2/vpn_auto_subnet/vpn-auto-subnet.jinja [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  name: {{ env["deployment"] }}-default-route
+  properties:
+    network: $(ref.{{ env["deployment"] }}-network.selfLink)
+    nextHopVpnTunnel: $(ref.{{ env["deployment"] }}-tunnel.selfLink)
+    priority: 100
+    destRange: 0.0.0.0/0
+    tags:
+      - {{ properties["routeTag"] }}
+
+
```
Comment (aljim) [inline examples/v2/vpn_auto_subnet/vpn-auto-subnet.jinja]: Remove last line

File: examples/v2/vpn_auto_subnet/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Cloud VPN with Auto Subnet Networks
+
+Templated Cloud VPN deployment with an auto subnet network
+
+## Prerequisites
+- Install [gcloud](https://cloud.google.com/sdk)
+- [Cloud VPN requirements](https://cloud.google.com/compute/docs/vpn/overview#requirements)
+- Create a project, setup billing, enable requisite APIs in [Google Cloud Console](https://console.cloud.google.com/)
```
Comment (aljim) [inline examples/v2/vpn_auto_subnet/README.md]: Split in two lines, to have less than 80 char per line

File: examples/v2/vpn_auto_subnet/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# Cloud VPN with Auto Subnet Networks
+
+Templated Cloud VPN deployment with an auto subnet network
+
+## Prerequisites
+- Install [gcloud](https://cloud.google.com/sdk)
+- [Cloud VPN requirements](https://cloud.google.com/compute/docs/vpn/overview#requirements)
```
Comment (aljim) [inline examples/v2/vpn_auto_subnet/README.md]: Split line to have less than 80 chars per line

**Comment by crcsmnky** [comment]
All changes look good, anything else needed to merge?

**Pull request merged**